src/policy/concrete.rs

```diff
@@ -1104,20 +1117,20 @@ fn with_huffman_tree<Pk: MiniscriptKey>(
 /// any one of the conditions exclusively.
 #[cfg(feature = "compiler")]
 fn generate_combination<Pk: MiniscriptKey>(
-    policy_vec: &Vec<Arc<Policy<Pk>>>,
+    policy_thresh: &Threshold<Arc<Policy<Pk>>>,
     prob: f64,
-    k: usize,
 ) -> Vec<(f64, Arc<Policy<Pk>>)> {
-    debug_assert!(k <= policy_vec.len());
-
     let mut ret: Vec<(f64, Arc<Policy<Pk>>)> = vec![];
-    for i in 0..policy_vec.len() {
-        let policies: Vec<Arc<Policy<Pk>>> = policy_vec
+    let k = policy_thresh.k();
+    for i in 0..policy_thresh.n() {
+        let policies: Vec<Arc<Policy<Pk>>> = policy_thresh
             .iter()
             .enumerate()
             .filter_map(|(j, sub)| if j != i { Some(Arc::clone(sub)) } else { None })
             .collect();
-        ret.push((prob / policy_vec.len() as f64, Arc::new(Policy::Threshold(k, policies))));
+        if let Ok(thresh) = Threshold::new(k, policies) {
+            ret.push((prob / policy_thresh.n() as f64, Arc::new(Policy::Thresh(thresh))));
+        }
```

Comment:
This looks like a change in logic but if you check the singled callsite for `generate_combination` you can see that this will not result in a logic change.

Reply:
The original code effectively used `new_unchecked`. If this really always results in a valid threshold (not immediately obvious from the code but I assume we would've noticed if this were wrong :)) I think we should use an `expect()` rather than an `if let Ok(..)` without any other branch.

Reply:
Can do, I might re-visit the call site too and see if there is any way to make it obvious that we can't hit the `expect`.